tabular/tests/unittests/resource_allocation/test_gpu_assignment.py

```diff
@@ -0,0 +1,490 @@
+        self._default_resources = default_resources
+        super().__init__(**kwargs)
+
+    def get_minimum_resources(self, **kwargs):
+        return self._minimum_resources
+
+    def _get_default_resources(self):
+        num_cpus = self._default_resources.get("num_cpus")
+        num_gpus = self._default_resources.get("num_gpus")
+        return num_cpus, num_gpus
+
+
+class DummyModel(DummyBaseModel):
+    pass
+
+
+class DummyBaggedModel(BaggedEnsembleModel):
+    pass
+
+
+def _prepare_data():
+    """Prepare test data for fold fitting strategy initialization."""
+    data = [[1, 10], [2, 20], [3, 30]]
+    df = pd.DataFrame(data, columns=["Number", "Age"])
+    label = "Age"
+    X = df.drop(columns=[label])
+    y = df[label]
+    return X, y
+
+
+def _construct_parallel_fold_strategy(
+    num_cpus=8,
+    num_gpus=4,
+    num_jobs=8,
+    num_folds_parallel=None,
+    model_base_minimum_resources=None,
+    model_base_default_resources=None,
+):
+    """
+    Construct a ParallelLocalFoldFittingStrategy instance for testing.
+
+    Parameters
+    ----------
+    num_cpus : int
+        Total CPUs available
+    num_gpus : int or float
+        Total GPUs available
+    num_jobs : int
+        Total number of fold jobs
+    num_folds_parallel : int, optional
+        Number of folds to train in parallel. Defaults to num_jobs if not specified.
+    model_base_minimum_resources : dict, optional
+        Minimum resources required by the model
+    model_base_default_resources : dict, optional
+        Default resources for the model
+
+    Returns
+    -------
+    ParallelLocalFoldFittingStrategy
+        An instance of the fold fitting strategy for testing
+    """
+    if num_folds_parallel is None:
+        num_folds_parallel = num_jobs
+    if model_base_minimum_resources is None:
+        model_base_minimum_resources = {"num_cpus": 1, "num_gpus": 0}
+    if model_base_default_resources is None:
+        model_base_default_resources = {"num_cpus": 4, "num_gpus": 0}
+
+    dummy_model_base = DummyModel(
+        minimum_resources=model_base_minimum_resources,
+        default_resources=model_base_default_resources,
+        hyperparameters={}
+    )
+    dummy_bagged_ensemble_model = DummyBaggedModel(dummy_model_base, hyperparameters={})
+    train_data, test_data = _prepare_data()
+
+    args = dict(
+        num_jobs=num_jobs,
+        num_folds_parallel=num_folds_parallel,
+        model_base=dummy_model_base,
+        model_base_kwargs=dict(),
+        bagged_ensemble_model=dummy_bagged_ensemble_model,
+        X=train_data,
+        y=test_data,
+        X_pseudo=None,
+        y_pseudo=None,
+        sample_weight=None,
+        time_start=time.time(),
+        time_limit=None,
+        models=[],
+        oof_pred_proba=np.array([]),
+        oof_pred_model_repeats=np.array([]),
+        save_folds=True,
+        num_cpus=num_cpus,
+        num_gpus=num_gpus,
+    )
+    return ParallelLocalFoldFittingStrategy(**args)
```

Comment:
Lets actually make some of the more generic parameters here have defaults in the actual class so we don't need to specify so many things here in the test explicitly.

For example, `X_pseudo, y_pseudo, `sample_weight, time_start, time_limit, save_folds` can all be given default values equal to what you have here.